runtime/lua/vim/filetype.lua

```diff
@@ -1715,6 +1717,7 @@ local pattern = {
   ['org%.eclipse%..*%.prefs'] = 'jproperties',
   ['.*%.properties_.._.._.*'] = starsetf('jproperties'),
   ['[jt]sconfig.*%.json'] = 'jsonc',
+  ['[jJ]ustfile'] = 'just',
```

Comment:
I omitted the wildcard based on @zeertzjq's comment; I'll leave this open for a bit in case Vim adds a fixup commit which I can then include as N/A.